### ISSUE TITLE
Style PICTURE with 'display: contents' and SOURCE with 'display: none'

### ISSUE THREAD

**Issue by fantasai** [mozdevs/cssremedy#45]
[Testcase](http://software.hixie.ch/utilities/js/live-dom-viewer/?%3C!DOCTYPE%20html%3E%0A%3Cstyle%3E%0A%20%20body%20*%20%7B%20border%3A%20dotted%20red%3B%20%7D%0A%20%20source%20%7B%20border%3A%20double%20green%3B%20%7D%0A%20%20img%20%7B%20border%3A%20solid%20blue%3B%20height%3A%2030px%3B%20%7D%0A%20%20%2F*%20source%20%7B%20display%3A%20none%3B%20%7D%0A%20%20picture%20%7B%20display%3A%20contents%3B%20%7D%20*%2F%0A%3C%2Fstyle%3E%0A%0A%3Cpicture%3E%0A%20%20%3Csource%20srcset%3D%22https%3A%2F%2Fvia.placeholder.com%2F350x150%22%0A%20%20%20%20%20%20%20%20%20%20%20%20media%3D%22(max-width%3A%2020px)%22%3E%0A%20%20%3Csource%20srcset%3D%22https%3A%2F%2Finteractive-examples.mdn.mozilla.net%2Fmedia%2Fexamples%2Fsurfer-240-200.jpg%22%0A%20%20%20%20%20%20%20%20%20%20%20%20media%3D%22(min-width%3A%2020px)%22%3E%0A%20%20%3Cimg%20src%3D%22https%3A%2F%2Fwww.w3.org%2Fblog%2Fnews%2Ffiles%2F2018%2F05%2Fw3c-ab-300x300.jpg%22%3E%0A%3C%2Fpicture%3E%0A)

The picture element, being treated as an inline wrapper around the actual replaced element, prevents images with aspect ratios in participating properly in the CSS layout algorithms as images with aspect ratios. Also unless one specifically wants a wrapper, having an inline box wrapped around the image isn't really helping anyone. Also having a bunch of empty inlines representing each SOURCE element seems pretty unnecessary. Let's get rid of these boxes.

(Yay, a use case for 'display: contents' that isn't better served by subgrid!)

**Comment by tabatkins** [comment]
This should presumably be raised on HTML, no?

**Comment by jensimmons** [comment]
@tabatkins Yes, it should. And here, until they fix it and make this unnecessary... meanwhile, I will remember the cries of "WHAT?!?!?!! IT DOES WHAT?????" from @fantasai @frivoal Amelia, and crew (from CSSWG) for while.

**Comment by Sora2455** [comment]
Does this need to be added to `<track>` elements too?

**Comment by frivoal** [comment]
> Does this need to be added to <track> elements too?

Doesn't seem so. `<video>` and `<audio>` are themselves replaced elements, and their children are not part of the page, so even if you put a border on a `<track>` element or a `<source>` element inside a video, it will not show up. It's only inside picture that it is a problem

**Comment by Malvoz** [comment]
`<map>` as well?
